cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -0,0 +1,709 @@
+    if profiles is None:
+        return False
+
+    if profiles == []:
+        cli_tell(f"No profiles were loaded from '{file_path}'")
+        return True
+
+    cli_tell("[CTRL + D or CTRL + C] to finish")
+    all_successful = True
+    try:
+        for profile in profiles:
+            all_successful = all_successful and complete_profile(profile, profiles)
+            cli_tell()
+    except (EOFError, KeyboardInterrupt):
+        log.info("Operation interrupted")
+        cli_tell()
+
+    if not backup_file(file_path):
+        log.warning("Failed to create profiles backup")
+
+    if not save_profiles(file_path, profiles):
+        return False
+
+    cli_tell("Finished completing profiles")
+    return all_successful
+
+
+def complete_profile(profile: AzureProfile, profiles: List[AzureProfile]) -> bool:
+    """
+    Complete the profile if any information is missing
+    All interactions with user happen here and in cli_* functions
+    """
+
+    cli_tell(f"Profile name: {profile.name}")
+
+    # check if anything is missing at all
+    missing_fields = list_missing_required_fields(profile)
+    if not missing_fields:
+        cli_tell("Profile is complete")
+        return True
+
+    missing_fields_str = ", ".join(missing_fields)
+    cli_tell(f"The profile is missing following fields: {missing_fields_str}")
+    # avoid logging into Azure if only principal_secret is missing - it can't be retrieved from the account anyway
+    if missing_fields == ["principal_secret"]:
+        profile.principal_secret = find_secret(profile.principal_id, profiles) or cli_ask_secret()
+        cli_tell(f"Profile '{profile.name}' information completed: {missing_fields_str}")
+        return True
+
+    # filling any information other than just principal_secret requires logging into Azure Account
+    account = cli_azure_login_interactively(profile.subscription_id, profile.tenant_id)
+    if account is None:
+        return False
+
+    # fill account information
+    profile.subscription_id = account.subscription_id
+    profile.tenant_id = account.tenant_id
+
+    # fill service principal information
+    if profile.principal_id == "":
+        principal = cli_get_or_create_service_principal(profiles, account.subscription_id)
+        if principal is None:
+            return False
+        profile.principal_id = principal.id
+        profile.principal_secret = principal.secret
+    elif profile.principal_secret == "":
+        profile.principal_secret = find_secret(profile.principal_id, profiles) or cli_ask_secret()
+
+    # fill azure location information
+    if profile.location == "":
+        profile.location = cli_ask_azure_location()
+
+    # fill resource groups information
+    if not profile.resource_group_names:
+        profile.resource_group_names = cli_list_resource_groups(profile.location)
+
+    cli_tell(f"Profile '{profile.name}' information completed: {missing_fields_str}")
```

Comment:
Shouldn't you verify that the user gave you everything (as opposed to say empty values)?

Reply:
not easy to say; if the user decided to give empty value, then from their perspective the profile is complete